front/src/applications/stdcm/utils/stdcmComputeChartData.ts

```diff
@@ -0,0 +1,56 @@
+import type { TFunction } from 'i18next';
+
+import { preparePathPropertiesData } from 'applications/operationalStudies/utils';
+import type { StdcmPathProperties, StdcmResponse } from 'applications/stdcm/types';
+import type { TrainScheduleResult, RollingStockWithLiveries } from 'common/api/osrdEditoastApi';
+import formatPowerRestrictionRangesWithHandled from 'modules/powerRestriction/helpers/formatPowerRestrictionRangesWithHandled';
+import { updateChartSynchronizerTrainData } from 'modules/simulationResult/components/ChartSynchronizer/utils';
+import type { SpeedSpaceChartData } from 'modules/simulationResult/types';
+
+/**
+ * Computes chart data for a given stdcm response.
+ */
+const computeChartData = async ({
+  stdcmResponse,
+  stdcmTrainResult,
+  t,
+  rollingStock,
+  pathProperties,
+}: {
+  stdcmResponse: StdcmResponse;
+  stdcmTrainResult: TrainScheduleResult;
+  t: TFunction;
+  rollingStock: RollingStockWithLiveries;
+  pathProperties: StdcmPathProperties | null;
+}): Promise<SpeedSpaceChartData | null> => {
+  if (stdcmResponse.status !== 'success' || pathProperties === null) {
+    return null;
+  }
+  const { simulation, path: pathfindingResult, departure_time: departureTime } = stdcmResponse;
+  if (!simulation || simulation.status !== 'success' || !pathfindingResult || !departureTime) {
+    return null;
+  }
+  const formattedPathProperties = preparePathPropertiesData(
+    simulation.electrical_profiles,
+    pathProperties,
+    pathfindingResult,
+    stdcmTrainResult.path,
+    t
+  );
+  const formattedPowerRestrictions = formatPowerRestrictionRangesWithHandled({
+    selectedTrainSchedule: stdcmTrainResult,
+    selectedTrainRollingStock: rollingStock,
+    pathfindingResult,
+    pathProperties: formattedPathProperties,
+  });
+  updateChartSynchronizerTrainData(simulation, rollingStock, departureTime);
```

Comment:
The Chart Synchronizer has been removed in a PR of @emersion :wink: You can remove this line